perf(transformer/jsx): pre-allocate enough memory for `arguments` of `createElement`/`jsx` function

Dunqing opened the pull request:
This is a hot path for JSX transform, allocating sufficient memory can avoid reserving in `push`, every bit of optimization that adds up can be huge.

Dunqing commented:
* **#9915** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9915?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9915?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9915).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
<img width="758" alt="image" src="https://github.com/user-attachments/assets/c18e5b3e-5110-4319-9aa9-a7b4215b90af" />

overlookmotel reviewed (approved):
Cowabunga!

overlookmotel commented:
### Merge activity

* **Mar 21, 4:43 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Mar 21, 4:43 AM EDT**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Dunqing committed:
perf(transformer/jsx): pre-allocate enough memory for `arguments` of `createElement`/`jsx` function (#9915)

This is a hot path for JSX transform, allocating sufficient memory can avoid reserving in `push`, every bit of optimization that adds up can be huge.
